### PR TITLE
Reinstate ESLint, enforce linting better

### PR DESCRIPTION
**Pull request by rtibbles**
## Summary

Reinstates ESLint.

Also noticed that Prettier linting is not being enforced by the CI, so tried to fix that.

Will be adding in back ESLint CI enforcement too as I iterate on making sure the ESLint config actually fits our codebase.

### CODE REVIEW

**Comment by indirectlylit** [comment]
should we also address the eslint errors in this PR?

**Comment by indirectlylit** [comment]
fixes #2216

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
+var path = require('path');
+
+module.exports = {
+  env: {
+    es6: true,
+    browser: true,
+    node: true,
+  },
+  parserOptions: {
+    sourceType: 'module',
```
Comment (jonboiser) [inline .eslintrc.js:10]: the bytesForHumans module does use the ES2016 exponentiation operator, which is why ecmaVersion was set to 7 in the old config.

**Comment by rtibbles** [comment]
@indirectlylit I think I had better. I've had no luck with any autofixing as yet though, so it may be a slightly convoluted process!

File: .eslintrc.js (hunk 1 above)
Comment (rtibbles) [reply .eslintrc.js:10]: The latest version of Buble does support this, but we should make sure buble is updated, and also polyfill Array.prototype.includes which is the other thing that made it into the ES2016 spec.

**Comment by rtibbles** [comment]
Well... that was better than I was expecting in terms of merge conflicts.

**Comment by rtibbles** [comment]
Will rebase, and repush.

**Comment by rtibbles** [comment]
Rebased.

**Pull request merged**